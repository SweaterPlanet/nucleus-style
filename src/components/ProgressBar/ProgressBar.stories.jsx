import React from 'react'
import {
  withKnobs,
  boolean,
  number,
  select,
} from '@storybook/addon-knobs/react'
import ProgressBar from './index'

export default {
  title: 'ProgressBar',
  component: ProgressBar,
  decorators: [withKnobs],
}

const colorOptions = {
  aqua: 'aqua',
  black: 'black',
  blue: 'blue',
  green: 'green',
  paper: 'paper',
  purple: 'purple',
  red: 'red',
  slate: 'slate',
  white: 'white',
}

const valueOptions = {
  range: true,
  min: 0,
  max: 100,
  step: 1,
}

export function Default() {
  return (
    <ProgressBar
      dark={boolean('dark', false)}
      color={select('color', colorOptions, 'blue')}
      value={number('value', 20, valueOptions)}
    />
  )
}
