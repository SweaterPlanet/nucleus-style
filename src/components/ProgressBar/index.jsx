import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import theme from '../../theme'

const Bar = styled.div`
  width: 100%;
  height: 8px;
  background-color: ${theme.colors.paper.base};

  ${props =>
    props.dark &&
    `
		background-color: ${theme.colors.white};
	`}
`

const Meter = styled.div`
  width: ${props => props.value}%;
  height: 8px;
  
	${props =>
    props.color === 'aqua' &&
    `
      background-color: ${theme.colors.aqua.base};
    `}  

  ${props =>
    props.color === 'black' &&
    `
      background-color: ${theme.colors.black};
    `} 


  ${props =>
    props.color === 'blue' &&
    `
      background-color: ${theme.colors.blue.base};
    `}  

  ${props =>
    props.color === 'green' &&
    `
      background-color: ${theme.colors.green.base};
    `}

  ${props =>
    props.color === 'paper' &&
    `
      background-color: ${theme.colors.paper.base};
    `} 

  ${props =>
    props.color === 'purple' &&
    `
      background-color: ${theme.colors.purple.base};
    `}

  ${props =>
    props.color === 'red' &&
    `
      background-color: ${theme.colors.red.base};
    `} 

  ${props =>
    props.color === 'slate' &&
    `
      background-color: ${theme.colors.slate.base};
    `} 

  ${props =>
    props.color === 'white' &&
    `
      background-color: ${theme.colors.white};
    `} 
`

function ProgressBar(props) {
  return (
    <Bar>
      <Meter value="20" color="blue"></Meter>
    </Bar>
  )
}

export default ProgressBar
