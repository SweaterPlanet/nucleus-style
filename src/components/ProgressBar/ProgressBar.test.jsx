import React from 'react'
import { mount, shallow } from 'enzyme'

import ProgressBar from '.'

describe('ProgressBar', () => {
  it('renders correctly', () => {
    const test = shallow(<ProgressBar />)
    expect(test).toMatchSnapshot()
  })

  it('respects the color prop', () => {
    const testProp = 'aqua'
    const test = mount(<ProgressBar color={testProp} />)
    expect(test.prop('color')).toBe(testProp)
  })

  it('respects the dark prop', () => {
    const test = mount(<ProgressBar dark />)
    expect(test.prop('dark')).toBe(true)
  })

  it('respects the value prop', () => {
    const testProp = 30
    const test = mount(<ProgressBar value={testProp} />)
    expect(test.prop('value')).toBe(testProp)
  })
})
