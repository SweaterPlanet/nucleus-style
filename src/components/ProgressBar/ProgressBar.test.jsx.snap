// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar renders correctly 1`] = `ShallowWrapper {}`;
